src/sentry/middleware/ratelimit.py

```diff
@@ -15,80 +15,92 @@
             rate_limit_cond = (
-                request.rate_limit_metadata.rate_limit_type != RateLimitType.NOT_LIMITED
+                rate_limit_metadata.rate_limit_type != RateLimitType.NOT_LIMITED
                 if ENFORCE_CONCURRENT_RATE_LIMITS
-                else request.rate_limit_metadata.rate_limit_type == RateLimitType.FIXED_WINDOW
+                else rate_limit_metadata.rate_limit_type == RateLimitType.FIXED_WINDOW
             )
             if rate_limit_cond:
-                request.will_be_rate_limited = True
                 enforce_rate_limit = getattr(view_func.view_class, "enforce_rate_limit", False)
                 if enforce_rate_limit:
-                    return HttpResponse(
+                    response = HttpResponse(
                         {
                             "detail": DEFAULT_ERROR_MESSAGE.format(
-                                limit=request.rate_limit_metadata.limit,
-                                window=request.rate_limit_metadata.window,
+                                limit=rate_limit_metadata.limit,
+                                window=rate_limit_metadata.window,
                             )
                         },
                         status=429,
                     )
+                    self.add_headers(response, rate_limit_metadata)
+                    return response
+
         except Exception:
             logging.exception("Error during rate limiting, failing open. THIS SHOULD NOT HAPPEN")
 
-    def process_response(self, request: Request, response: Response) -> Response:
-        try:
-            rate_limit_metadata: RateLimitMeta | None = getattr(
-                request, "rate_limit_metadata", None
-            )
-            if rate_limit_metadata:
-                response["X-Sentry-Rate-Limit-Remaining"] = rate_limit_metadata.remaining
-                response["X-Sentry-Rate-Limit-Limit"] = rate_limit_metadata.limit
-                response["X-Sentry-Rate-Limit-Reset"] = rate_limit_metadata.reset_time
-                response[
-                    "X-Sentry-Rate-Limit-ConcurrentRemaining"
-                ] = rate_limit_metadata.concurrent_remaining
-                response[
-                    "X-Sentry-Rate-Limit-ConcurrentLimit"
-                ] = rate_limit_metadata.concurrent_limit
-            if hasattr(request, "rate_limit_key") and hasattr(request, "rate_limit_uid"):
-                finish_request(request.rate_limit_key, request.rate_limit_uid)
-        except Exception:
+        # Hit the endpoint
+        request.rate_limit_metadata = rate_limit_metadata
+        response = self.get_response(request)
+
+        # Process the response
+        self.add_headers(response, request.rate_limit_metadata)
+        finish_request(rate_limit_key, rate_limit_uid)
+        return response
+
+    def add_headers(self, response, rate_limit_metadata):
+        if not rate_limit_metadata or type(response) not in (Response, HttpResponse):
```

Comment:
In this case, if the `rate_limit_metadata` is not present this will log an exception. But the rate_limit_metadata may not be present simply because the endpoint is not supposed to be rate limited. Exceptions should only be logged when they are actually exceptions.